### PR TITLE
Enhancements for adding filter with ChannelId on condition

### PR DESCRIPTION
**Pull request by kedarkhaire**
Updated Config form of Teams Settings form.
Changes for condition based query parameter addition

### CODE REVIEW

**Comment by kedarkhaire** [comment]
Fixes #947

File: modules/apigee_edge_teams/src/Entity/Controller/TeamController.php [hunk 1]
```diff
@@ -233,4 +235,39 @@ public function delete(string $entity_id): EntityInterface {
...
+          $queryparam = [
+            'filter' => 'channelId=' . $channelid,
+          ];
+        }
+        else {
+          $defaultChannelId = TeamAliasForm::originalChannelId();
+          $queryparam = [
+            'filter' => 'channelId=' . $defaultChannelId,
+          ];
+        }
```
Comment (shishir-intelli) [inline modules/apigee_edge_teams/src/Entity/Controller/TeamController.php]: Use ternary operator instead to reduce code to fewer lines.

File: modules/apigee_edge_teams/src/Entity/Controller/TeamController.php [hunk 2]
```diff
@@ -233,4 +235,39 @@ public function delete(string $entity_id): EntityInterface {
     return $entity;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function getEntities(PagerInterface $pager = NULL, string $key_provider = 'id'): array {
```
Comment (shishir-intelli) [inline modules/apigee_edge_teams/src/Entity/Controller/TeamController.php:241]: Pagination not working after inheriting `getEntities()`.

File: modules/apigee_edge_teams/src/Entity/Controller/TeamController.php [hunk 3]
```diff
@@ -233,4 +235,39 @@ public function delete(string $entity_id): EntityInterface {
...
+          ];
+        }
+        $entities = $this->decorated()->getEntities($pager, $key_provider, $queryparam);
+      }
+      else {
+        $entities = $this->decorated()->getEntities($pager, $key_provider);
+      }
+    }
+    else {
+      $entities = $this->decorated()->getEntities($pager, $key_provider);
```
Comment (shishir-intelli) [inline modules/apigee_edge_teams/src/Entity/Controller/TeamController.php]: `$this->decorated()->getEntities` called thrice.

**Review by shishir-intelli — CHANGES_REQUESTED** [review]

File: modules/apigee_edge_teams/src/Entity/Controller/TeamController.php (hunk 2 above)
Comment (kedarkhaire) [reply modules/apigee_edge_teams/src/Entity/Controller/TeamController.php:241]: Hi @shishir-intelli  The required changes are made.

File: modules/apigee_edge_teams/src/Entity/Controller/TeamController.php (hunk 1 above)
Comment (kedarkhaire) [reply modules/apigee_edge_teams/src/Entity/Controller/TeamController.php]: Hi @shishir-intelli  The required changes are made.

File: modules/apigee_edge_teams/src/Entity/Controller/TeamController.php (hunk 3 above)
Comment (kedarkhaire) [reply modules/apigee_edge_teams/src/Entity/Controller/TeamController.php]: Hi @shishir-intelli  The required changes are made.

File: src/Entity/Controller/CachedPaginatedEntityListingControllerTrait.php [hunk 4]
```diff
@@ -55,7 +55,7 @@ public function getEntities(PagerInterface $pager = NULL, string $key_provider =
       }
     }
 
-    $entities = $this->decorated()->getEntities($pager, $key_provider);
+    $entities = $this->decorated()->getEntities($pager, $key_provider, $queryparam);
```
Comment (shishir-intelli) [inline src/Entity/Controller/CachedPaginatedEntityListingControllerTrait.php]: If we are not caching it, then no need of any changes on this file, your changes are anyway not reaching here.

**Review by shishir-intelli — APPROVED** [review]
LGTM

**Pull request merged**